pepper-apis/src/main/java/org/broadinstitute/ddp/service/UserDeleteService.java

```diff
@@ -0,0 +1,416 @@
+     * be deleted first).
+     *
+     * @param handle     JDBI handle
+     * @param user       User to be deleted
+     * @param whoDeleted information about a person who deleted a user (for example it can be an operatorGuid)
+     * @param comment    some additional comments if needed
+     */
+    public void fullDelete(Handle handle, User user, String whoDeleted, String comment) throws IOException {
+        delete(handle, user, whoDeleted, comment, true);
+    }
+
+    private void delete(Handle handle, User user, String whoDeleted, String comment, boolean fullDelete) throws IOException {
+        LOG.info("User {} deletion is STARTED. guid:{}", fullDelete ? "FULL" : "SIMPLE", user.getGuid());
+        if (fullDelete) {
+            checkBeforeDelete(handle, user);
+        }
+        try {
+            deleteUserData(handle, user, fullDelete);
+            LOG.warn("User {} deletion is completed SUCCESSFULLY. "
+                            + "guid:{}, hruid:{}, e-mail:{}, name:{} {}, dob:{}\n"
+                            + "Who deleted: {}\n"
+                            + "Comment: {}\n",
+                    fullDelete ? "FULL" : "SIMPLE",
+                    user.getGuid(),
+                    user.getHruid(),
+                    user.getEmail(),
+                    user.getProfile() != null ? objToStr(user.getProfile().getFirstName()) : "",
+                    user.getProfile() != null ? objToStr(user.getProfile().getLastName()) : "",
+                    user.getProfile() != null ? objToStr(user.getProfile().getBirthDate()) : "",
+                    objToStr(whoDeleted),
+                    objToStr(comment));
+        } catch (Throwable e) {
+            LOG.error(format(EXCEPTION_MESSAGE_PREFIX__ERROR + e.getMessage(), user.getGuid()), e);
+            throw new DDPException(format(EXCEPTION_MESSAGE_PREFIX__ERROR + e.getMessage()));
+        }
+    }
+
+    private void checkBeforeDelete(Handle handle, User user) {
+        if (hasGovernedUsers(handle, user.getGuid())) {
+            throw new DDPException(format(EXCEPTION_MESSAGE_PREFIX__ERROR + "the user has governed users", user.getGuid()));
+        }
+
+        // check if user refers to revisions
+        long[] revisionIds = handle.attach(JdbiRevision.class).findByUserId(user.getId());
+        if (revisionIds.length > 0) {
+            throw new DDPException(format(EXCEPTION_MESSAGE_PREFIX__ERROR + "the user has references to a revision history",
+                    user.getGuid()));
+        }
+    }
+
+    private void deleteUserData(Handle handle, User user, boolean fullDelete) throws IOException {
+        UserCollectedData userCollectedData = new UserCollectedData();
+
+        if (fullDelete) {
+            deleteKitRequests(handle, user);
+            deleteUserStudyLegacyData(handle, user);
+            deleteUserAnnouncement(handle, user);
+            deleteActivityInstanceStatusByOperator(handle, user);
+            deleteFileUpload(handle, user);
```

Comment:
I would classify fileupload deletion as part of removing activity data and therefore make it part of "normal" deletion.

Reply:
Moved to "normal" deletion.